src/Indice.Features.Cases.App/src/app/app.links.ts

```diff
@@ -1,40 +1,65 @@
-import { map, tap } from 'rxjs/operators';
+import { map, switchMap, toArray } from 'rxjs/operators';
 import { AuthService } from '@indice/ng-auth';
 import { IAppLinks, NavLink } from '@indice/ng-components';
-import { Observable, of, ReplaySubject } from 'rxjs';
+import { Observable, ReplaySubject, combineLatest, of } from 'rxjs';
+import { Params } from '@angular/router';
+import { CaseTypeService } from './core/services/case-type.service';
+import { Icons } from './shared/icons';
 
 export class AppLinks implements IAppLinks {
 
-    private _main: ReplaySubject<NavLink[]> = new ReplaySubject<NavLink[]>(1);
-
-    constructor(private authService: AuthService) {
-        this.authService.user$.pipe(
-            map(user => {
-                const headerMenu = [
-                    new NavLink('Αρχική', '/dashboard', true),
-                    new NavLink('Υποθέσεις', '/cases', true)
-                ]
-                if (this.authService.isAdmin()) {
-                    headerMenu.push(new NavLink('Διαχείριση Υποθέσεων', '/case-types', true));
-                } else {
-                    headerMenu.push(new NavLink('Ειδοποιήσεις', '/notifications', true));
-                }
-                return headerMenu;
-            }),
-            tap(navLinks => this._main.next(navLinks))
-        ).subscribe();
-    }
-
-    public public: Observable<NavLink[]> = of([]);
-    public profileActions: Observable<NavLink[]> = of([]);
-
-    public main: Observable<NavLink[]> = this._main.asObservable();
-
-    public profile: Observable<NavLink[]> = of([
-        new NavLink('Αποσύνδεση', '/logout', false)
-    ]);
-
-    public legal: Observable<NavLink[]> = of([]);
-    public brand: Observable<NavLink[]> = of([]);
-}
+  private _main: ReplaySubject<NavLink[]> = new ReplaySubject<NavLink[]>(1);
+
+  constructor(
+    private authService: AuthService,
+    private _caseTypeService: CaseTypeService
+  ) {
+    const isAdmin$ = this.authService.user$.pipe(
+      switchMap(user => of(this.authService.isAdmin()))
+    );
+
+    const caseTypeMenuItems$ = isAdmin$.pipe(
+      switchMap(isAdmin => isAdmin ? this._caseTypeService.getCaseTypeMenuItems() : of([]))
```

Comment:
δεν ισχύει αυτό που γράφεις. Θέλω να βλέπω τα case types menu item ακόμη και αν δεν είμαι admin.

Reply:
Αν αντι αυτου γραψω:     ```
const caseTypeMenuItems$ = this.authService.user$.pipe(
      switchMap(() => this._caseTypeService.getCaseTypeMenuItems())
    );
```
τοτε με πεταει εξω και μου λεει να ξανακανω log in. Θεωρητικα θα επρεπε να πιανει γιατι περιμενω να παρει user profile πριν κανω get.